doc/SONIC_Test_Ingress_Discard_All_HLD.md

```diff
@@ -0,0 +1,606 @@
+    - [Test case #6](#test-case-6)
+    - [Test case #7](#test-case-7)
+    - [Test case #8](#test-case-8)
+    - [Test case #9](#test-case-9)
+    - [Test case #10](#test-case-10)
+    - [Test case #11](#test-case-11)
+    - [Test case #12](#test-case-12)
+    - [Test case #13](#test-case-13)
+    - [Test case #14](#test-case-14)
+    - [Test case #15](#test-case-15)
+    - [Test case #16](#test-case-16)
+
+#### [Note]
+Proposal to implement more extendent test cases on phase 2 when https://github.com/Azure/SONiC/blob/master/doc/drop_counters/drop_counters_HLD.md will be implemented.
+
+#### Overview
+The purpose is to test "ingress_discard_all" counter triggers on receiving specific packets.
+The "ingress_discard_all" counter counts all discard events. This counter counts concurrently with other discard counters.
+The test assumes all necessary configuration are already pre-configured on the SONIC switch before test runs.
+Destination IP address of the injected packet must be routable to ensure packet should be routed but was dropped.
+
+#### Scope
+The purpose of the test is testing of "ingress_discard_all" counter triggering on SONIC system, making sure that specific traffic drops correctly, according to sent packet and configured packet discards.
+Supported topologies:
+```
+t0
+t1
+t1-lag
+ptf32
+```
+
+#### Related DUT CLI commands
+Command to check "ingress_discard_all" counter value:
+```show interfaces counters```
+
+Check field:
+```RX_DRP```
+
+#### General test flow
+- Select two pairs of PTF and DUT ports considering topology: PTF_PORT[1] <--->DUT_PORT[1], PTF_PORT[2] <--->DUT_PORT[2]
+- Clear counters. Use CLI command "sonic-clear counters"
+
+- For buffer counters verification:
+	- [TODO]
+- For other counter verification:
+	- Inject 1 packet into PTF_PORT[1]
+- Check "ingress_discard_all" counter incremented:
+	- For buffer counters verification:
+		- [TODO]
+	- For other counter verification:
+		- By 1
+- Additional verification:
+    [Add additional verifications for L2, L3, ACL and buffers counters in phase 2. When https://github.com/Azure/SONiC/blob/master/doc/drop_counters/drop_counters_HLD.md will be implemented]
+- Check the packet was dropped by sniffing traffic on PTF_PORT[2]
+
+
+#### Run test
+```
+py.test --inventory ../ansible/inventory --host-pattern [DEVICE] --module-path ../ansible/library/ --testbed [DEVICE-TOPO] --testbed_file ../ansible/testbed.csv --junit-xml=./target/test-reports/ --show-capture=no --log-cli-level debug -ra -vvvvv ingress_discard/test_ingress_discard.py
+```
+
+#### Test cases
+Each test case will be additionally validated by the loganalyzer utility.
+
+Each test case will run specific traffic to trigger specific discard reasone.
+
+Pytest fixture - "ptfadapter" is used to construct and send packets.
+
+After packet is sent using source port index, test framework waits for specific packet did not appear, due to ingress packet drop, on one of the destination port indices.
```

Comment:
is there a timer for that? how much time do you wait?

Reply:
Test framework can wait for example 5 seconds, to ensure there is no expected packet.